File: src/javaCode/boj/bfs_dfs/_1260.java
package javaCode.boj.bfs_dfs;

import java.io.*;
import java.util.ArrayList;
import java.util.Iterator;
import java.util.StringTokenizer;

public class _1260 {

    private static BufferedWriter bw;
    private static BufferedReader br;
    private static StringTokenizer st;
    private static boolean[] visited;
    private static ArrayList<Node>[] arr;

    private static class Node {
        int n;
        int d;

        public Node(int n, int d) {
            this.n = n;
            this.d = d;
        }
    }

    public static void main(String[] args) throws IOException {
        bw = new BufferedWriter(new OutputStreamWriter(System.out));
        br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        visited = new boolean[N + 1];
        arr = new ArrayList[N + 1];

        for (int i = 0; i < N + 1; i++) {
            arr[i] = new ArrayList<>();
        }

        for (int i = 0; i < N - 1; i++) {
            st = new StringTokenizer(br.readLine());
            int n1 = Integer.parseInt(st.nextToken());
            int n2 = Integer.parseInt(st.nextToken());
            int d = Integer.parseInt(st.nextToken());

            arr[n1].add(new Node(n2, d));
            arr[n2].add(new Node(n1, d));
        }

        for (int i = 1; i < N + 1; i++) {
            visited = new boolean[N + 1];
            bw.write(dfs(i, visited, 0) + "\n");
        }
        bw.flush();
        bw.close();
    }

    static int dfs(int s, boolean[] visited, int sum) {

        Iterator<Node> it = arr[s].iterator();
        visited[s] = true;
        int result = sum;

        while (it.hasNext()) {
            Node node = it.next();
            if (!visited[node.n]) {
                result += dfs(node.n, visited, sum + node.d);
            }
        }
        return result;
    }
}

File: src/kotlinCode/boj/_1260.kt
package kotlinCode.boj

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.LinkedList


private val br = System.`in`.bufferedReader()
private val bw = System.out.bufferedWriter()

fun main() = with(br) {

    val (n, m, v) = readLine().split(" ").map { it.toInt() }
    val array = Array(n + 1) { mutableListOf<Int>() }

    repeat(m) {
        val (v1, v2) = readLine().split(" ").map { it.toInt() }
        array[v1].add(v2)
        array[v2].add(v1)
    }

    array.forEach {
        it.sort()
    }

    var visited = BooleanArray(n + 1)
    dfs(array, visited, v)
    visited = BooleanArray(n + 1)
    bw.write("\n")
    bfs(array, visited, v)

    bw.flush()
    bw.close()
}

fun dfs(array: Array<MutableList<Int>>, visited: BooleanArray, v: Int) {
    visited[v] = true
    bw.write("$v ")
    array[v].forEach {
        if (!visited[it]) {
            dfs(array, visited, it)
        }
    }
}

fun bfs(array: Array<MutableList<Int>>, visited: BooleanArray, v: Int) {
    var queue = LinkedList<Int>()
    queue.add(v)
    visited[v] = true

    while (!queue.isEmpty()) {
        var nv = queue.poll()
        bw.write("$nv ")
        array[nv].forEach {
            if (!visited[it]) {
                queue.add(it)
                visited[it] = true
            }
        }
    }
}